src/plot_api/plot_api.js

```diff
@@ -2156,7 +2156,11 @@ Plotly.animate = function(gd, frameOrGroupNameOrFrameList, animationOpts) {
     gd = helpers.getGraphDiv(gd);
 
     if(!Lib.isPlotDiv(gd)) {
-        throw new Error('This element is not a Plotly plot: ' + gd);
+        throw new Error(
+            'This element is not a Plotly plot: ' + gd + '. It\'s likely that you\'ve failed ' +
```

Comment:
Agreed. I think it'd be nice in general to beef up the error messages, particularly when the input looks fishy—like passing an object where an an array is expected and instead of failing it just doesn't get any props from it.